pkg/cloud/libvirt/client/client.go

```diff
@@ -0,0 +1,87 @@
+package client
+
+import (
+	libvirt "github.com/libvirt/libvirt-go"
+	providerconfigv1 "github.com/openshift/cluster-api-provider-libvirt/pkg/apis/libvirtproviderconfig/v1alpha1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// CreateDomainInput specifies input parameters for CreateDomain operation
+type CreateDomainInput struct {
+	// DomainName is name of domain to be created
+	DomainName string
+
+	// IgnKey is name of existing volume with ignition config (DEPRICATED)
+	IgnKey string
+
+	// Ignition configuration to be injected during bootstrapping
+	Ignition *providerconfigv1.Ignition
+
+	// CloudInit configuration to be run during bootstrapping
+	CloudInit *providerconfigv1.CloudInit
+
+	// VolumeName of volume to be added to domain definition
+	VolumeName string
+	// PoolName of pool where VolumeName volume is located
```

Comment:
nit: consistently add a blank line between fields. Or not. Prefer the former as it easier to read and you already have many fields in this form.